lib/spack/spack/test/concretize.py

```diff
@@ -525,3 +526,24 @@ def test_regression_issue_7941(self):
         t.concretize()
 
         assert s.dag_hash() == t.dag_hash()
+
+    @pytest.mark.parametrize('abstract_specs, checklist', [
```

Comment:
Can you add another test to show that two packages that would naturally concretize to different specs for a shared  dependency share the dependency spec when concretized together?

Reply:
@becker33 That's already the case: the versions of both `callpath` and `dyninst` are not the latest available, and without concretizing specs together e.g. `mpileaks` should have `callpath@1.0` and `dyninst@8.2`. Or were you suggesting a more complex scenario than this?

Reply:
Can you make that more explicit?

```
# Concretize mpileaks to confirm it uses newer callpath/dyninst to ensure we
# actually test co-concretization.
mpileaks_spec = Spec('mpileaks').concretized()
assert mpileaks_spec.satisfies('^dyninst@8.1.2:')
assert mpileaks_spec.satisfies('^callpath@0.10:')

... Then go into the rest of the test

Reply:
Sure, I added an initial test to set the baseline rather than concretizing `mpileaks` as part of each test. Let me know if that's fine with you.

Reply:
@alalazo did you commit your new test? I don't see it here.

Reply:
Yes, it's:

https://github.com/spack/spack/blob/de551bc517faaaae3def1fed7aa094193c446eb7/lib/spack/spack/test/concretize.py#L531-L533

Reply:
Oh got it, wasn't parsing the arguments properly.